### PR TITLE
Cleanup and update for go modules & new Sentry client

### PR DESCRIPTION
**Pull request by krisreeves-stripe**
#### Summary / Motivation
Does a few things:
- Fixes a bug where we didn't return when falling back from JSON to raw text (TBD: add a test)
- Updates to go modules from dep
- Switches from raven-go to sentry-go
- Exposes some methods in austerity to share tests and logic with overridden sampling behavior (future: make this some kind of configurable interface instead of overriding it with custom behavior)
- Sets a semver version number

#### Test plan
Unit tests

#### Rollout/monitoring/revert plan
Merge, then tag with an actual version for module use

### CODE REVIEW

File: filters/timeprefix.go [hunk 1]
```diff
@@ -19,15 +19,15 @@ type TimePrefixFilter struct {
 
 // FilterLine prepends the current time, in square brackets with a separating
 // space, to the provided log line.
-func (f *TimePrefixFilter) FilterLine(line string) string {
+func (f TimePrefixFilter) FilterLine(line string) string {
```
Comment (asf-stripe) [inline filters/timeprefix.go:22]: I was worried that passing the filter by value would be slow, but it's a bool & a string value, so should be fine.

Comment (krisreeves-stripe) [reply filters/timeprefix.go:22]: The filter was unusable without this signature; switching it to pointer receivers is a bigger change for the future maybe :(

The performance effects you mention could potentially be a thing but it's a thing that's been for some time if so

Comment (krisreeves-stripe) [reply filters/timeprefix.go:22]: Actually... this is that Go thing about interfaces and pointer receivers that I don't quite get yet. The test this caused a problem with is https://github.com/stripe/unilog/pull/42/files#diff-2845aa72f8c25965ab21a0741a0b0d2dR297

**Comment by krisreeves-stripe** [comment]
re-r? @asf-stripe

**Pull request merged**